src/Sarif/Writers/PrereleaseCompatibilityTransformer.cs

```diff
@@ -0,0 +1,553 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+using System;
+using System.Collections.Generic;
+using System.Text.RegularExpressions;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace Microsoft.CodeAnalysis.Sarif.Writers
+{
+    public static class PrereleaseCompatibilityTransformer
+    {
+        // As the SARIF Technical Committee develops the SARIF specification, it
+        // releases incremental versions of the schema, with SemVer versions such as
+        // "2.0.0-csd.2.beta.2018-10-10". To avoid having to update the version strings
+        // in the test files every time this happens, we replace the version
+        // in the test file with the current version after reading the file
+        // into memory.
+        private const string VersionPropertyPattern = @"""version""\s*:\s*""[^""]+""";
+        private static readonly Regex s_VersionRegex = new Regex(VersionPropertyPattern, RegexOptions.Compiled);
+
+        private const string SchemaPropertyPattern = @"""\$schema""\s*:\s*""[^""]+""";
+        private static readonly Regex s_SchemaRegex = new Regex(SchemaPropertyPattern, RegexOptions.Compiled);
+
+        public static string UpdateToCurrentVersion(string prereleaseSarifLog, bool forceUpdate = false, Formatting formatting = Formatting.None)
+        {
+            bool modifiedLog = false;
+            JObject sarifLog = JObject.Parse(prereleaseSarifLog);
+
+            // Some tests update the semantic version to current for non-updated content. For this situation, we 
+            // allow the test code to force a transform, despite the fact that the provided version doesn't call for it.
+            string version = forceUpdate ? "2.0.0" : (string)sarifLog["version"];
```

Comment:
>"2.0.0" [](start = 43, length = 7)

Suggestion: use `VersionConstants.AssemblyVersion` here and on Line 38.

Reply:
Won't take this suggestion, because some critical logical would change abruptly if that constant changes (as it should be able to do).

---
In reply to: [226758312](https://github.com/Microsoft/sarif-sdk/pull/1114#discussion_r226758312) [](ancestors = 226758312)